apps/fft/fft_aot_test.cpp

```diff
@@ -0,0 +1,288 @@
+#include <cmath>
+#include <iostream>
+#include <iomanip>
+
+#include "generated/fft_forward_r2c.h"
+#include "generated/fft_inverse_c2r.h"
+#include "generated/fft_forward_c2c.h"
+#include "generated/fft_inverse_c2c.h"
+
+namespace {
+const float kPi = 3.14159265358979310000f;
+
+const size_t kSize = 16;
+}
+
+// Make a buffer_t for real input to the FFT.
+buffer_t real_buffer(float *storage, int32_t y_size = kSize) {
+    buffer_t buf = {0};
+
+    buf.host = (uint8_t *)storage;
+    buf.extent[0] = kSize;
+    buf.stride[0] = 1;
+    buf.extent[1] = y_size;
+    buf.stride[1] = kSize;
+    buf.elem_size = sizeof(float);
+
+    return buf;
+}
+
+// Make a buffer_t for complex input to the FFT.
+buffer_t complex_buffer(float *storage, int32_t y_size = kSize) {
+    buffer_t buf = {0};
+
+    buf.host = (uint8_t *)storage;
+    buf.extent[0] = 2;
```

Comment:
Also, what about using Tuples for this? ComplexExpr is just a thin wrapper around Tuple, it is a convenient way to work with complex numbers in Halide.